packages/shell-api/src/replica-set.spec.ts

```diff
@@ -26,6 +26,41 @@ function deepClone<T>(value: T): T {
   return JSON.parse(JSON.stringify(value));
 }
 
+
+type RetriableFn = (...fnArgs: any) => Promise<any>;
+
+function createRetriableFunc(rs: ReplicaSet, fn: keyof ReplicaSet, totalRetries = 12): RetriableFn {
```

Comment:
You can probably make this a bit more precise by using:

```suggestion
function createRetriableFunc<F extends keyof ReplicaSet>(rs: ReplicaSet, fn: F, totalRetries = 12): ReplicaSet[F] {
```

instead of having a `RetriableFn` which is equivalent to just “any function”.

You’d need to use `let reconfigWithRetry: ReplicaSet['reconfig'];` below then

Reply:
This is definitely what I was looking for. Thanks 👍